src/utils/api-server.tsx

```diff
@@ -64,6 +88,431 @@ export const getApiServerBaseUrl = (): string => {
   return 'https://' + apiHost + ':' + apiPort + getProxyUrl() + '/api/v1';
 };
 
+type SignatureSubFormType = {
+  name: string;
+  signature: Signature;
+};
+
+export const SignatureSubForm: FC<SignatureSubFormType> = ({
+  name,
+  signature,
+}) => {
+  const authToken = useContext(AuthContext);
+  const [formValues, setFormvalues] = useState<Record<string, string>>({});
+  const update = (name: string, value: string) => {
+    const newFormValues = { ...formValues };
+    newFormValues[name] = value;
+    setFormvalues(newFormValues);
+    resetLoginMutation();
+  };
+
+  const {
+    data: response,
+    mutate: execRequest,
+    isError,
+    isSuccess,
+    isLoading,
+    reset: resetLoginMutation,
+  } = useJolokiaServiceExecBrokerOperation({});
+  return (
+    <FormGroup label={name}>
+      <p>{signature.desc}</p>
+      {signature.args.map((arg, item) => {
+        return (
+          <FormGroup label={arg.name + ' ' + arg.type} key={item}>
+            <TextInput
+              value={formValues ? formValues[arg.name] : ''}
+              onChange={(value) => update(arg.name, value)}
+              aria-label={arg.name}
+            />
+          </FormGroup>
+        );
+      })}
+      <ActionGroup>
+        <Button
+          variant="primary"
+          onClick={() => {
+            if (!formValues) {
+              return;
+            }
+            execRequest({
+              jolokiaSessionId: authToken,
+              requestBody: {
+                signature: {
+                  name: name,
+                  args: signature.args.map((arg) => {
+                    const ret: OperationArgument = {
+                      value: formValues
+                        ? formValues[arg.name]
+                          ? formValues[arg.name]
+                          : ''
+                        : '',
+                      type: arg.type,
+                    };
+                    return ret;
+                  }),
+                },
+              },
+            });
+          }}
```

Comment:
Hey @lavocatt , nice work!!
Just a small suggestion: instead of adding the lengthy function inside the button component, could you please create a separate function to handle the onClick logic and keep the component clean?